core/src/main/java/org/elasticsearch/discovery/local/LocalDiscovery.java

```diff
@@ -305,6 +285,11 @@ public DiscoveryStats stats() {
     }
 
     @Override
+    public DiscoverySettings getDiscoverySettings() {
+        return discoverySettings;
+    }
+
```

Comment:
Not sure I follow - do you mean have the methods/logic in DiscoverySettings folded into that new base class Discovery , which all discoveries need to inherit from?  If that's the case, I'm doubtful - it's nice to have all the settings encapsulated so we can pass DiscoverySettings to other stuff like the publish action. However, it doesn't contain all settings so we're not there either. Long story shot - setting should be picked up but I rather not do it in this change....